lib/scheduler.js

```diff
@@ -29,7 +29,7 @@ function Scheduler (options) {
     var self = this;
 
     self.options = {};
-    self.options.frameworkName = options.frameworkName.replace(/ /g, "-") || "mesos-framework." + uuid.v4();
+    self.options.frameworkName = (options.frameworkName || "mesos-framework." + uuid.v4()).replace(/ /g, "-");
```

Comment:
I think it'd be ok if it fails then... The default value can't have a space char even IMO, so I guess it doesn't really make sense to add the replace there.